gemini/sample-apps/fashion-trends-prediction/app/utilities.py

```diff
@@ -0,0 +1,187 @@
+        str: The sentence with duplicate words removed.
+
+    """
+    return " ".join(dict.fromkeys(sentence.split()))
+
+
+def nav_page(page_name, timeout_secs=3):
+    """Navigates to a page in the Streamlit app.
+
+    Args:
+        page_name (str): The name of the page to navigate to.
+        timeout_secs (int, optional): The number of seconds to wait for the page to load. Defaults to 3.
+
+    """
+    nav_script = """
+        <script type="text/javascript">
+            function attempt_nav_page(page_name, start_time, timeout_secs) {
+                var links = window.parent.document.getElementsByTagName("a");
+                for (var i = 0; i < links.length; i++) {
+                    if (links[i].href.toLowerCase().endsWith("/" + page_name.toLowerCase())) {
+                        links[i].click();
+                        return;
+                    }
+                }
+                var elasped = new Date() - start_time;
+                if (elasped < timeout_secs * 1000) {
+                    setTimeout(attempt_nav_page, 100, page_name, start_time, timeout_secs);
+                } else {
+                    alert("Unable to navigate to page '" + page_name + "' after " + timeout_secs + " second(s).");
+                }
+            }
+            window.addEventListener("load", function() {
+                attempt_nav_page("%s", new Date(), %d);
+            });
+        </script>
+    """ % (
+        page_name,
+        timeout_secs,
+    )
+    html(nav_script)
+
+
+# To remove background from an image
+def removeBg(url, userName):
+    """Removes the background from an image.
+
+    Args:
+        url (str): The URL of the image to remove the background from.
+        userName (str): The name of the user who uploaded the image.
+
+    Returns:
+        str: The path to the image with the background removed.
+
+    """
+    input_path = url
+    actual_img_path = "" + userName + "$.png"
+    urllib.request.urlretrieve(input_path, actual_img_path)
+
+    # Store path of the output image in the variable output_path
+    output_path = "" + userName + "#.png"
+
+    # Processing the image
+    input = ImagePIL.open(actual_img_path)
+
+    # Removing the background from the given Image
+    output = remove(input)
+
+    # Saving the image in the given path
+    output.save(output_path)
+
+    return actual_img_path, output_path
+
+
+def get_base64_of_bin_file(png_file):
+    """Converts a binary file to a base64 encoded string.
+
+    Args:
+        png_file (str): The path to the binary file.
+
+    Returns:
+        str: The base64 encoded string of the binary file.
+
+    """
+    with open(png_file, "rb") as f:
+        data = f.read()
+    return base64.b64encode(data).decode()
+
+
+def render_svg(svg):
+    """Renders an SVG string.
+
+    Args:
+        svg (str): The SVG string to render.
+
+    """
+    """Renders the given svg string."""
+    b64 = base64.b64encode(svg.encode("utf-8")).decode("utf-8")
```

Comment:
Some of this is duplicated from other places. For SVGs, you might be able to display them directly in the code as well.

Reply:
This function is updated now